Feature/thumbnail

brunoCo3ta opened the pull request:
New Thumbnail web-component

brunoCo3ta committed:
Create thumbnail.js

New Web component Thumbnail

receive filename and call web api to get the image content

brunoCo3ta committed:
Add new Thumbnail Component

brunoCo3ta committed:
Update thumbnail.js

brunoCo3ta committed:
Update thumbnail.js

List Dispay Issues
Show Error Messages

brunoCo3ta committed:
Change Upload logic Add StoruyBook and readme

Change Upload logic
Add StoruyBook and readme

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.stories.js:
```diff
@@ -0,0 +1,69 @@
+    	return Promise.resolve()
+}
+
+function UplodaFile(){
```
typo

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,365 @@
+      
+     
+      this._container = document.createElement("div");
+      this._container.id = "Thubnailbox";
```
typo

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,365 @@
+        entity: null,
+        lastCodeValue: null,
+        fileToUpload: null,
+        lastpath: null,
```
lastpath? shouldn't be "Path"?p

gsferreira committed:
Add prettier

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,365 @@
+    }
+  
+    set context(newValue) {
+      if (newValue) {
```
invert if

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,365 @@
+  
+    set state(newValue) {
+      this._settings.state = newValue;
+      if (this._settings.state != null &&   this._settings.lastCodeValue !== this._settings.state._code ) {
```
invert if

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,365 @@
+  
+      const apiClient = this._settings.context.createApiHttpClient();
+
+      apiClient.doGetFile(url, "image/jpeg")
```
Does this work with PNG for example?

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,365 @@
+                .then(()=>{
+                  hideErrorMessage();
+                  let newValue = ""; 
+                  newValue = `${this._settings.entity}/${this._settings.lastCodeValue}/${file.name}`;
```
change let to const

gsferreira commented on webcomponents/web-components/Thumbnail/readme.md:
```diff
@@ -0,0 +1,8 @@
+| Property | Description                     | Default Value |
+|----------|---------------------------------|---------------|
+| isReadOnly | Sets the elements as read-only (not allowing to add and remove images) |    true      |
+| value | The current image: a string with TenantName/Entity Identifier/ File name |        |
```
Add an example

gsferreira commented on webcomponents/web-components/Thumbnail/readme.md:
```diff
@@ -0,0 +1,8 @@
+### Customization
+| Property | Description                     | Default Value |
+|----------|---------------------------------|---------------|
+| isReadOnly | Sets the elements as read-only (not allowing to add and remove images) |    true      |
```
add or remove

brunoCo3ta committed:
Resolve PR Comments

brunoCo3ta committed:
Merge branch 'feature/thumbnail' of https://github.com/OMNIALowCode/omnia3-samples into feature/thumbnail

brunoCo3ta committed:
Change Documentation

gsferreira committed:
Format code

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,328 @@
+  constructor() {
+    super();
+
+    this.downloadFile = this.downloadFile.bind(this);
```
This function doesn't exist so the bind will fail

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,328 @@
+    const reader = new FileReader();
+    reader.readAsDataURL(blob);
+
+    let avatar_container = document.querySelector(`#Thumbnail_${this._settings.lastCodeValue}.avatar-container`);
```
change to const

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,328 @@
+    const url = `${this.endpoint(originalCode)}/${fileName}`;
+    const apiClient = this._settings.context.createApiHttpClient();
+    apiClient.doDelete(url).then(response => {
+      this._settings.files = this._settings.files.filter(f => f.name !== file);
```
suggestion: extract a function to handle the response

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,328 @@
+    this._settings.entity = fileNameSplit.length > 1 ? fileNameSplit[0] : this._settings.entity;
+    const url = `${this.endpoint(originalCode)}/${fileName}`;
+    const apiClient = this._settings.context.createApiHttpClient();
+    apiClient.doDelete(url).then(response => {
```
shouldn't we implement a catch?

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,328 @@
+    this._settings.language = this._settings.context.getLanguageTranslator().language;
+  }
+
+  set state(newValue) {
```
please review this setter. Is getting difficult to read.

brunoCo3ta committed:
PR resolve comments

brunoCo3ta committed:
Merge branch 'feature/thumbnail' of https://github.com/OMNIALowCode/omnia3-samples into feature/thumbnail

brunoCo3ta committed:
Format Code

brunoCo3ta committed:
Binding Issues

brunoCo3ta committed:
typo fix

gsferreira committed:
review readme

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.stories.js:
```diff
@@ -0,0 +1,68 @@
+
+    component.context = context;
+    component.value = text('Value', 'File Path');
+    component.isReadOnly = boolean('Is read only', true);
```
If the user changes the knob to false, he will see a message of "undefined"

brunoCo3ta committed:
PR Comments and Resolve Issue

Resolve Knob issue
Change logic to delete file when user change the identifier
Refactoring

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,380 @@
+  }
+
+  //Functions
+  AppendReadOnlyElements(newAvatarContainer, newFeedBackContainer) {
```
review casing

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,380 @@
+    this._feedBackElement.appendChild(newFeedBackContainer);
+  }
+
+  AppendEditElements(newAvatarContainer, newFeedBackContainer) {
```
review casing

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,380 @@
+    this._feedBackElement.appendChild(newFeedBackContainer);
+  }
+
+  ResetElementIdentifiers(currentAvatarContainer, currentfeedBackelement) {
```
review casing

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,380 @@
+  AppendReadOnlyElements(newAvatarContainer, newFeedBackContainer) {
+    this._container.id = `ThumbnailBox_${this._settings.lastCodeValue}`;
+    this._container.appendChild(newAvatarContainer);
+    this._feedBackElement.appendChild(newFeedBackContainer);
```
typo: _feedBackElement -> _feedbackElement

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,380 @@
+      this.AppendReadOnlyElements(newAvatarContainer, newFeedBackContainer);
+    } else {
+      var currentAvatarContainer = document.querySelector("div[id^='Thumbnail_']");
+      var currentfeedBackelement = document.querySelector("div[id^='ThumbnailError_']");
```
typo: currentfeedBackelement  -> currentFeedbackElement

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.js:
```diff
@@ -0,0 +1,380 @@
+    this.uploadFile = this.uploadFile.bind(this);
+
+    this._settings = {
+      baseUrl: `${window.location.protocol}//${window.location.host}/api/v1/`,
```
not being used

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.stories.js:
```diff
@@ -0,0 +1,69 @@
+import { document, console } from 'global';
```
console not used

gsferreira commented on webcomponents/web-components/Thumbnail/thumbnail.stories.js:
```diff
@@ -0,0 +1,69 @@
+import { document, console } from 'global';
+import { storiesOf } from '@storybook/html';
+import { boolean, text } from '@storybook/addon-knobs';
```
text not used

brunoCo3ta committed:
Clean Code
